### PR TITLE
Fix makesecret api, remove getsharedsecret

### PR DESCRIPTION
**Pull request by rustyrussell**
Make makesecret API match datastore API, and remove getsharesecret.

### CODE REVIEW

**Comment by adi2011** [comment]
I think you'll also have to remove the `getsharedsecret.schema.json` file, apart from that LGTM...

**Review by adi2011 — CHANGES_REQUESTED** [review]
I think you'll also have to remove the `getsharedsecret.schema.json` file, apart from that LGTM...

**Comment by rustyrussell** [comment]
> I think you'll also have to remove the `getsharedsecret.schema.json` file, apart from that LGTM...

Good catch!  Done.

**Comment by rustyrussell** [comment]
Removed file as suggested by @adi2011  and also removed duplicate makesecret test...

**Review by adi2011 — APPROVED** [review]

File: tests/test_misc.py [hunk 1]
```diff
@@ -2723,3 +2731,12 @@ def test_torv2_in_db(node_factory):
     l1.stop()
     l1.db_manip("UPDATE peers SET address='3fyb44wdhnd2ghhl.onion:1234';")
     l1.start()
+
+
+def test_makesecret(node_factory):
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply tests/test_misc.py]: Yep, I've deleted this now!

**Comment by adi2011** [comment]
ACK b658d29

**Review by cdecker — APPROVED** [review]
ACK b658d299f184bc8a38329f3c0b413b5c6eda1a64

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/5430/commits/b658d299f184bc8a38329f3c0b413b5c6eda1a64

**Pull request merged**